bot/handlers/linked_roles_test.go

```diff
@@ -10,9 +10,30 @@ import (
 	"github.com/labstack/echo-contrib/session"
 	"github.com/labstack/echo/v4"
 	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
 	"golang.org/x/oauth2"
 )
 
+func TestHandleLinkedRolesRedirect(t *testing.T) {
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+
+	e.Use(session.Middleware(sessions.NewCookieStore([]byte("secret"))))
+
+	oauthConfig := &oauth2.Config{}
+	logger, _ := zap.NewDevelopment()
+
+	e.GET("/", func(c echo.Context) error {
+		return HandleLinkedRolesRedirect(logger, c, oauthConfig)
+	})
+
+	e.ServeHTTP(rec, req)
+
+	// Assertions
```

Comment:
@jules please add assertions to test that a session is present with a non-nil state as expected